code/iaas/logic/src/main/resources/META-INF/cattle/core-process/spring-core-processes-context.xml

```diff
@@ -144,4 +144,25 @@
     <!-- Cluster host map -->
     <process:process name="clusterhostmap.create" resourceType="clusterHostMap" start="requested" transitioning="creating" done="created" />
     <process:process name="clusterhostmap.remove" resourceType="clusterHostMap" start="created,creating" transitioning="removing" done="removed" />
+
+    <!-- Service Discovery Environment -->
+    <process:process name="environment.create" resourceType="environment" start="requested" transitioning="activating" done="active" />
+    <process:process name="environment.activateservices" resourceType="environment" start="active" transitioning="updating-active" done="active" />
+    <process:process name="environment.deactivateservices" resourceType="environment" start="active" transitioning="updating-active" done="active" />
```

Comment:
done state for this is confusing since it's still "active".  I had a similar question above too with the transition from "active" to "active".  I'm wondering whether activateservices and deactivateservices should be regular actions rather than actual transition states.  Sort of like the resourceActions "addhost" and "removehost" in cluster.json and ClusterAddHostActionHandler.java